### PR TITLE
Messaging

### PR DESCRIPTION
**Pull request by sebgrebe**
* Adds messaging users and group organisers  
* Toggled of unless IS_MESSAGING_ENABLED is set to true

### CODE REVIEW

**Comment by sebgrebe** [comment]
QA  

- [x] User can toggle on/off messaging  
- [x] Toggle on/off only shows when messaging feature toggle is on  
- [x] User can message other users  
- [x] Links in email to user are correct  
- [x] User Messaging hidden and not possible if recipient is unapproved
- [x] User Messaging hidden and not possible if  recipient has toggled messaging off
- [x] User Messaging hidden and not possible if messaging is generally toggled off.
- [x] User Messaging hidden and not possible if sender not logged in.
- [x] User messaging not possible if sender is unapproved
- [x] Unapproved user cannot message groups or other users  
- [x] User can message group without group email but with group organiser email  
- [x] User can message group with group email  
- [x] Links in email to group are correct
- [x] Group Messaging hidden and not possible if group has no emails associated and no organisers
- [x] Group Messaging hidden and not possible if group has no group emails associated and messaging not enabled
- [x] Group messaging hidden and not possible if only email is organisers and that organiser is not approved or has messaging not enabled  
- [x] group messaging hidden and not possible if sender not logged in

**Comment by sebgrebe** [comment]
To do to switch on (once email has been sent out):
- [ ] Set IS_MESSAGING_ENABLED env variable to true

File: .env [hunk 1]
```diff
@@ -9,3 +9,4 @@ DJANGO_SETTINGS_MODULE=eahub.config.settings
 LEAN_MANAGERS="Local Manager=leanmanager@example.com"
 GROUPS_EMAIL=groups@example.com
 ADMINS="Local Admin=admin@example.com"
+FEEDBACK_URL="https://feedback.eahub.org"
```
Comment (viktor-yunenko) [inline .env]: @sebgrebe, do you think it should be an env variable?

Comment (sebgrebe) [reply .env]: I thought that's what you suggested on the previous MR on this? (I might misremember this though). 
If not, what would you suggest instead?

Comment (viktor-yunenko) [reply .env]: maybe you're talking about URL reversing? sadly there's no way to reverse that one

In general I don't recommend to store in envs anything except secrets and environment variables, it has downsides eg as cluttering and inaccessibility. With django when something must be configurable it's better to move it out into db using a specific package as https://github.com/lazybird/django-solo

Comment (sebgrebe) [reply .env]: Added django solo and removed env var

File: requirements.in [hunk 2]
```diff
@@ -51,3 +51,4 @@ pytest
 pip-tools
 faker
 model_bakery
+django-solo
```
Comment (viktor-yunenko) [inline requirements.in:54]: FYI that won't be a dev package I think?

**Review by viktor-yunenko — APPROVED** [review]
Thank you! I wanted to add django-solo to this project for a while now 😅

**Comment by viktor-yunenko** [comment]
I'll also try to find a bit of time to move other envs to the site config as well, since I had too many incidents with env management and on other django project and really would like to prevent this from happening

**Pull request merged**